app/controllers/hosts_controller.rb

```diff
@@ -102,6 +102,10 @@ def create
     else
       load_vars_for_ajax
       offer_to_overwrite_conflicts
+      # add errors from other sources (e.g. Katello) to base errors
+      errors = @host.errors.full_messages | @host.errors[:base]
+      @host.errors.delete(:base)
+      @host.errors.add(:base, errors.join(', '))
```

Comment:
The controller should not change the errors. Shouldn't Katello try to explicitly flag the field that triggered the error so it's highlighted as wrong? And if anything, the view should read the base errors if they're there.

We already have some of that processing:
https://github.com/theforeman/foreman/blob/d309c1f37590fe4df1c7f3992d614e3319a4ed95/app/controllers/application_controller.rb#L311-L325

Another consideration: the API controller won't have the same validation this way.